plugins/workspace-plugin/src/workspace-projects-manager.ts

```diff
@@ -33,37 +34,56 @@ export function handleWorkspaceProjects(pluginContext: theia.PluginContext, proj
   });
 }
 
+type Project = cheApi.workspace.devfile.Project | cheApi.workspace.ProjectConfig;
+
 abstract class WorkspaceProjectsManager {
```

Comment:
Do we seriously still support non-devfile workspaces?

Reply:
I create the issue for the refactoring workspace creation related logic: https://github.com/eclipse/che/issues/19082

Reply:
Please see https://github.com/eclipse/che/issues/15249#issuecomment-780369425